crates/bevy_ecs/src/component.rs

```diff
@@ -813,7 +813,7 @@ impl ComponentHooks {
 #[derive(Debug, Clone)]
 pub struct ComponentInfo {
     id: ComponentId,
-    descriptor: ComponentDescriptor,
+    descriptor: Arc<ComponentDescriptor>,
```

Comment:
It's unfortunate to have to wrap ordinary `ComponentDescriptor`s in an `Arc` even though most of them aren't behind a lock.  That means we have to do a `clone()` on every lookup!  

Would it work to have `get_descriptor` return an `atomicow::CowArc<ComponentDescriptor>`?  That would let you return either a borrowed value or an owned `Arc`, and it would deref to `ComponentDescriptor`.